pkg/routeagent_driver/handlers/calico/ippool_handler.go

```diff
@@ -129,10 +129,6 @@ func (h *calicoIPPoolHandler) TransitionToGateway() error {
 }
 
 func (h *calicoIPPoolHandler) Uninstall() error {
-	if !h.isGateway.Load() {
```

Comment:
The [bug report ](https://github.com/submariner-io/submariner/issues/2764) says that Calico IPPools are not deleted as part of Uninstall. 

@yboaron, this PR would ensure that every route-agent tries to delete the Calico IPPools. Even without this fix, the route-agent on the Gateway node should have deleted the Calico IPPools right?

Reply:
@sridhargaddam , also thought that with the original code the RA pod that runs on GW node should delete the Calico IPPools.
 
I think that the root cause is that 'TransitionToGateway' event not sent to Calico handler when we run the RA uninstall daemonset pod.

Reply:
The Calico IPPool handler in route-agent uses the same handler framework as used by the other handlers, so ideally it should be notified of `TransitionToGateway` event. Are you seeing this [bug](https://github.com/submariner-io/submariner/issues/2764) consistently in your setup or was it a one-off case?

Reply:
I hit this bug consistently on my env,  even after I manually [edit RA clusterrole to enable deletecollection ](https://github.com/submariner-io/submariner-operator/pull/2856)

Reply:
>The Calico IPPool handler in route-agent uses the same handler framework as used by the other handlers, so ideally it should be notified of TransitionToGateway event

Q: Do you expect that Calico handler (or any other handlers) should be notified of TransitionToGateway event even for RA Uninstall DaemonSet ?

Reply:
As discussed over slack, the TransitionToGateway would be fired during regular Gateway migrations which would update the isGateway status to True. However, during Uninstall,  an all-together new uninstall Pod would be created and this will not be aware of the Gateway status. Good catch Yossi.